### PR TITLE
Feature/player hand evaluation

### LINKED ISSUE #19: Add hand evaluation functionality to Player class

### PR DESCRIPTION
**Pull request by meszszi**
Resolves #19

### CODE REVIEW

File: server/pom.xml [hunk 1]
```diff
@@ -37,6 +37,12 @@
             <version>5.4.2</version>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>com.marcinmoskala</groupId>
```
Comment (CrunchyNomNom) [inline server/pom.xml]: 😍

Comment (meszszi) [reply server/pom.xml]: role model

File: server/src/main/kotlin/core/gameflow/Player.kt [hunk 2]
```diff
@@ -36,4 +39,12 @@ data class Player(
     }
 
     fun withCards(newCards: List<Card>): Player = this.copy(holeCards = newCards)
+
+    /* Creates the best possible hand out of player's hole cards and community cards. */
+    fun hand(communityCards: List<Card>): PokerHand {
+        assert(holeCards.isNotEmpty())
+        val cards = communityCards + holeCards
+        val possibleHands = cards.toSet().combinations(5).map { makeHand(it.toList()) }
+        return possibleHands.reduce { acc, next -> if (acc > next) acc else next }
```
Comment (wujuu) [inline server/src/main/kotlin/core/gameflow/Player.kt]: nice

File: server/src/test/kotlin/core/gameflow/PlayerTest.kt [hunk 3]
```diff
@@ -147,4 +152,31 @@ class PlayerTest {
...
+
+        val properHand = TwoPair(listOf(
+                Card(CardRank.ACE, CardSuit.SPADES),
+                Card(CardRank.TWO, CardSuit.CLUBS),
+                Card(CardRank.TWO, CardSuit.DIAMONDS),
+                Card(CardRank.SIX, CardSuit.CLUBS),
+                Card(CardRank.SIX, CardSuit.HEARTS)
+        ))
+
+        assert(player.hand(communityCards).compareTo(properHand) == 0)
```
Comment (wujuu) [inline server/src/test/kotlin/core/gameflow/PlayerTest.kt]: Not writing tests is nice, but you know what is nicer? A working application.

Comment (meszszi) [reply server/src/test/kotlin/core/gameflow/PlayerTest.kt]: :(

**Pull request merged**